platform/Sqlite.roc

```diff
@@ -0,0 +1,494 @@
+# dec = realSqlDecoder Ok
+
+# These are the same decoders as above but Nullable.
+# If the sqlite field is `Null`, they will return `Null`.
+
+Nullable a : [NotNull a, Null]
+
+nullable_str : Str -> SqlDecode (Nullable Str) UnexpectedTypeErr
+nullable_str = decoder \val ->
+    when val is
+        String s -> Ok (NotNull s)
+        Null -> Ok Null
+        _ -> to_unexpected_type_err val
+
+nullable_bytes : Str -> SqlDecode (Nullable (List U8)) UnexpectedTypeErr
+nullable_bytes = decoder \val ->
+    when val is
+        Bytes b -> Ok (NotNull b)
+        Null -> Ok Null
+        _ -> to_unexpected_type_err val
+
+nullable_int_decoder : (I64 -> Result a err) -> (Str -> SqlDecode (Nullable a) [FailedToDecodeInteger err]UnexpectedTypeErr)
+nullable_int_decoder = \cast ->
+    decoder \val ->
+        when val is
+            Integer i -> cast i |> Result.map NotNull |> Result.mapErr FailedToDecodeInteger
+            Null -> Ok Null
+            _ -> to_unexpected_type_err val
+
+nullable_real_decoder : (F64 -> Result a err) -> (Str -> SqlDecode (Nullable a) [FailedToDecodeReal err]UnexpectedTypeErr)
+nullable_real_decoder = \cast ->
+    decoder \val ->
+        when val is
+            Real r -> cast r |> Result.map NotNull |> Result.mapErr FailedToDecodeReal
+            Null -> Ok Null
+            _ -> to_unexpected_type_err val
+
+nullable_i64 : Str -> SqlDecode (Nullable I64) [FailedToDecodeInteger []]UnexpectedTypeErr
+nullable_i64 = nullable_int_decoder Ok
+
+nullable_i32 : Str -> SqlDecode (Nullable I32) [FailedToDecodeInteger [OutOfBounds]]UnexpectedTypeErr
+nullable_i32 = nullable_int_decoder Num.toI32Checked
+
+nullable_i16 : Str -> SqlDecode (Nullable I16) [FailedToDecodeInteger [OutOfBounds]]UnexpectedTypeErr
+nullable_i16 = nullable_int_decoder Num.toI16Checked
+
+nullable_i8 : Str -> SqlDecode (Nullable I8) [FailedToDecodeInteger [OutOfBounds]]UnexpectedTypeErr
+nullable_i8 = nullable_int_decoder Num.toI8Checked
+
+nullable_u64 : Str -> SqlDecode (Nullable U64) [FailedToDecodeInteger [OutOfBounds]]UnexpectedTypeErr
+nullable_u64 = nullable_int_decoder Num.toU64Checked
+
+nullable_u32 : Str -> SqlDecode (Nullable U32) [FailedToDecodeInteger [OutOfBounds]]UnexpectedTypeErr
+nullable_u32 = nullable_int_decoder Num.toU32Checked
+
+nullable_u16 : Str -> SqlDecode (Nullable U16) [FailedToDecodeInteger [OutOfBounds]]UnexpectedTypeErr
+nullable_u16 = nullable_int_decoder Num.toU16Checked
+
+nullable_u8 : Str -> SqlDecode (Nullable U8) [FailedToDecodeInteger [OutOfBounds]]UnexpectedTypeErr
+nullable_u8 = nullable_int_decoder Num.toU8Checked
+
+nullable_f64 : Str -> SqlDecode (Nullable F64) [FailedToDecodeReal []]UnexpectedTypeErr
+nullable_f64 = nullable_real_decoder Ok
+
+nullable_f32 : Str -> SqlDecode (Nullable F32) [FailedToDecodeReal []]UnexpectedTypeErr
+nullable_f32 = nullable_real_decoder (\x -> Num.toF32 x |> Ok)
+
+# TODO: Mising Num.toDec and Num.toDecChecked
+# nullable_dec = nullable_real_decoder Ok
```

Comment:
Can we use [`Str.toDec : Str -> Result Dec [InvalidNumStr]`](https://www.roc-lang.org/builtins/Str#toDec)